ltt_test/main.c

```diff
@@ -62,144 +59,116 @@ int main(void) {
+            last_message_millis = millis();
         }
-        
-        if (!rcvb_is_empty())
-        {
+
+        if (!rcvb_is_empty()) {
             can_msg_t msg;
             rcvb_pop_message(&msg);
             serialize_can_msg(&msg);
         }
-        
+
         //send any queued CAN messages
         txb_heartbeat();
+        
+        // clear watchdog timer
+        CLRWDT();
+        
+        // reset radio if no message received for over 1 minute
+        if (millis() - last_message_millis > RESET_RADIO_ms){
+            // reset radio
+            SET_RADIO_POWER(0);
+            uint32_t wait = millis();
+            while (millis() - wait < 50);    // wait for 50 ms
+            SET_RADIO_POWER(1);
+            last_message_millis = millis();
+        }
+        
+        // manually turn on/off radio through CAN message
+        if (set_radio_power){
+            SET_RADIO_POWER(radio_power);
+            set_radio_power = false;
+        }
     }
 }
 
 uint8_t high_fq_data_counter = 0;
+
 static void can_msg_handler(const can_msg_t *msg) {
     uint16_t msg_type = get_message_type(msg);
-    
+
     // A little hacky, but convenient way to filter out the high speed stuff 
     //(not including altitude cause we need that)
-    if (msg_type >= MSG_SENSOR_ACC && msg_type <= MSG_SENSOR_MAG){
+    if (msg_type >= MSG_SENSOR_ACC && msg_type <= MSG_SENSOR_MAG) {
         //while we want to discard most of the messages, we want to send them once in a while to alow checking that everything is alive
         // we use a prime number to avoid aliasing ensure that every message gets a chance to be sent
         high_fq_data_counter++;
         if (high_fq_data_counter >= HIGH_SPEED_MSG_DIVIDER) {
             high_fq_data_counter = 0;
             rcvb_push_message(msg);
         }
+    } else {
+        rcvb_push_message(msg);
     }
-    else {
-        // Don't send the message over uart if the bus is down and it's not from us
-        if (get_board_unique_id(msg) == BOARD_UNIQUE_ID || bus_powered) {
-            // Send the message over UART
-            rcvb_push_message(msg);
-        }
-    }
-    
+
     // ignore messages that were sent from this board
     if (get_board_unique_id(msg) == BOARD_UNIQUE_ID) {
         return;
     }
-    
+
     int dest_id = -1;
 
     switch (msg_type) {
         case MSG_ACTUATOR_CMD:
-            if (get_actuator_id(msg) == CANBUS) {
-                req_bus_powered = get_req_actuator_state(msg) == ACTUATOR_CLOSED;
+            if (get_actuator_id(msg) == ACTUATOR_RADIO) {
+                if (get_req_actuator_state(msg) == ACTUATOR_OFF && radio_power){
+                    set_radio_power = true;
+                    radio_power = false;
+                }
+                else if (get_req_actuator_state(msg) == ACTUATOR_ON && !radio_power){
+                    set_radio_power = true;
+                    radio_power = true;
+                }
```

Comment:
might be a good idea to remove check on radio_power so the command always set the power to desired one regardless of the previous state, for cases where there is a bug or a hyperintelligent cosmic proton messes with the state.